openslides_backend/presenter/search_deleted_models.py

```diff
@@ -0,0 +1,89 @@
+from typing import Any, List
+
+import fastjsonschema
+from datastore.shared.util import DeletedModelsBehaviour
+
+from ..permissions.management_levels import OrganizationManagementLevel
+from ..permissions.permission_helper import has_organization_management_level
+from ..shared.exceptions import PermissionDenied
+from ..shared.filters import And, Filter, FilterOperator, Or
+from ..shared.schema import schema_version
+from .base import BasePresenter
+from .presenter import register_presenter
+
+schema = fastjsonschema.compile(
+    {
+        "$schema": schema_version,
+        "type": "object",
+        "properties": {
+            "collection": {"type": "string"},
+            "filter_string": {"type": "string"},
+            "meeting_id": {"type": "integer", "minimum": 1},
+        },
+        "required": ["collection", "filter_string"],
+        "additionalProperties": False,
+    }
+)
+
+
+search_fields = {
+    "assignment": ["title"],
+    "motion": ["number", "title"],
+    "user": [
+        "username",
+        "first_name",
+        "last_name",
+        "title",
+        "pronoun",
+        "structure_level",
+        "number",
+        "email",
+    ],
+}
+
+
+@register_presenter("search_deleted_models")
+class SearchDeletedModels(BasePresenter):
+    """
+    Searches all deleted models of the given collection for a given filter string.
+    """
+
+    schema = schema
+
+    def get_result(self) -> Any:
+        if not has_organization_management_level(
+            self.datastore, self.user_id, OrganizationManagementLevel.SUPERADMIN
+        ):
+            raise PermissionDenied("You are not a superadmin")
+
+        collection = self.data["collection"]
+        meeting_id = self.data["meeting_id"]
+        filters: List[Filter] = []
+        for search_field in search_fields[collection]:
```

Comment:
This raises an error with collections not in search field.
You should think about checking the collections in the schema